src/nodedef.cpp

```diff
@@ -1069,6 +1070,36 @@ content_t CNodeDefManager::allocateDummy(const std::string &name)
 }
 
 
+void CNodeDefManager::unallocateName(const std::string &name)
+{
+	// Pre-condition
+	assert(name != "");
+
+	// Erase name from name ID mapping
+	content_t id = CONTENT_IGNORE;
+	if (m_name_id_mapping.getId(name, id)) {
+		m_name_id_mapping.eraseName(name);
+		m_name_id_mapping_with_aliases.erase(name);
+	}
+
+	// Erase node content from all groups it belongs to
+	for (std::map<std::string, GroupItems>::iterator i = m_group_to_items.begin();
+		i != m_group_to_items.end() ; ++i) {
+		GroupItems &items = i->second;
+		for (GroupItems::iterator j = items.begin(); j != items.end(); j++) {
+			if (j->first == id) {
+				items.erase(j);
+				j--;
+			}
+		}
+
+		// Check if group is empty
+		if (items.size() == 0)
+			m_group_to_items.erase(i->first);
```

Comment:
Because std::map::erase() invalidates any iterators referencing any of the elements erased.
See: http://en.cppreference.com/w/cpp/container/map/erase